scripts/shell/bt_nus_shell.py

```diff
@@ -1,84 +1,88 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019-2020 Nordic Semiconductor ASA
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+# flake8: noqa
 
 import logging
-import Queue
+import queue
 import time
 import socket
 import sys
-from thread import *
+from _thread import *
 import argparse
 
-HOST = ''   # Symbolic name meaning all available interfaces
-PORT = 8889 # Arbitrary non-privileged port
+from pc_ble_driver_py import config
+config.__conn_ic_id__ = "nrf52"
+
+from nordicsemi.dfu.dfu_transport_ble import DFUAdapter
+from pc_ble_driver_py.exceptions import NordicSemiException
+from pc_ble_driver_py.ble_driver import Flasher, BLEDriver, BLEDriverObserver, BLEConfig, BLEConfigConnGatt,\
+    BLEUUIDBase, BLEUUID, BLEAdvData, BLEGapConnParams
+from pc_ble_driver_py.ble_driver import ATT_MTU_DEFAULT
+from pc_ble_driver_py.ble_adapter import BLEAdapter, BLEAdapterObserver, EvtSync
+
+
+CFG_TAG = 1
 
 parser = argparse.ArgumentParser(description='BLE serial daemon.')
 parser.add_argument('--com', help='COM port name (e.g. COM110).',
-                    nargs='*', required=False, default="")
+                    nargs='*', required=True, default="")
 parser.add_argument('--snr', help='segger id.',
                     nargs=1, required=False, default="")
 parser.add_argument('--family', help='chip family.',
                     nargs=1, required=False, default="")
 parser.add_argument('--name', help='Device name (advertising name).',
-                    nargs='*', required=True, default=None)
-args = parser.parse_args()
-
-def ble_driver_init(conn_ic_id):
-    global BLEDriver, Flasher, DfuTransportBle
-    from pc_ble_driver_py import config
-    config.__conn_ic_id__ = conn_ic_id
+                    nargs='*', required=False, default=None)
+parser.add_argument('--port', help='Local port to use.',
+                    required=False, default=8889)
 
-ble_driver_init(args.family[0])
+args = parser.parse_args()
 
-from nordicsemi.dfu.dfu_transport_ble import DFUAdapter
-from pc_ble_driver_py.exceptions      import NordicSemiException, IllegalStateException
-from pc_ble_driver_py.ble_driver      import config,Flasher,BLEDriver, BLEDriverObserver, BLEEnableParams, BLEUUIDBase, BLEUUID, BLEAdvData, BLEGapConnParams, NordicSemiException
-from pc_ble_driver_py.ble_driver      import ATT_MTU_DEFAULT
-from pc_ble_driver_py.ble_adapter     import BLEAdapter, BLEAdapterObserver, EvtSync
+HOST = ''   # Symbolic name meaning all available interfaces
+PORT =  int(args.port)
 
-global nrf_sd_ble_api_ver
+# global nrf_sd_ble_api_ver
 nrf_sd_ble_api_ver = config.sd_api_ver_get()
-logger  = logging.getLogger(__name__)
+logger = logging.getLogger(__name__)
 
-class BLE_Serial(BLEDriverObserver, BLEAdapterObserver):
-    BASE_UUID   = BLEUUIDBase([0x6E, 0x40, 0x00, 0x00, 0xB5, 0xA3, 0xF3, 0x93,
+
+class BleSerial(BLEDriverObserver, BLEAdapterObserver):
+    BASE_UUID = BLEUUIDBase([0x6E, 0x40, 0x00, 0x00, 0xB5, 0xA3, 0xF3, 0x93,
```

Comment:
Unrelated changes like this make it harder to see the parts of this commit that are related to the python2 -> python3 conversion.

"Unrelated" because the BLE_Serial -> BleSerial rename and the whitespace change in initializing BASE_UUID are not required for converting to Python 3.